### PR TITLE
Update to MathOptInterface 0.5

### PR DESCRIPTION
**Pull request by tkoolen**
Supersedes #25.

### CODE REVIEW

**Comment by tkoolen** [comment]
`contlinear` tests were failing because we don't actually support changing the objective sense, but were reporting that we could.

**Comment by rdeits** [comment]
Aha, good catch

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,23 @@
-# Documentation: http://docs.travis-ci.com/user/languages/julia/
+## Documentation: http://docs.travis-ci.com/user/languages/julia/
```
[the remark this answers is not in the document]
Comment (tkoolen) [reply .travis.yml]: Fixed, must have commented stuff out and then not uncommented everything.

File: REQUIRE [hunk 2]
```diff
@@ -1,5 +1,5 @@
 BinDeps
 julia 0.6
-Compat
+Compat 0.47.0 # for @warn
```
[the remark this answers is not in the document]
Comment (tkoolen) [reply REQUIRE]: I disagree with this one. If new Compat features are used, the version bound should be updated, along with this comment.

File: test/runtests.jl [hunk 3]
```diff
@@ -2,6 +2,11 @@ using OSQP
 using Compat
 using Compat.Test, Compat.SparseArrays, Compat.LinearAlgebra, Compat.Random
 
+if isdefined(Random, :seed!)
```
[the remark this answers is not in the document]
Comment (tkoolen) [reply test/runtests.jl]: Done.

**Comment by tkoolen** [comment]
Thanks, @mlubin!

**Pull request merged**